cvat-ui/src/components/create-cloud-storage-page/location.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useState } from 'react';
+import Divider from 'antd/lib/divider';
+import Select from 'antd/lib/select';
+import { PlusCircleOutlined, QuestionCircleOutlined } from '@ant-design/icons';
+import Input from 'antd/lib/input';
+import Button from 'antd/lib/button';
+import Form from 'antd/lib/form';
+import notification from 'antd/lib/notification';
+import Tooltip from 'antd/lib/tooltip';
+
+const { Option } = Select;
+interface Props {
+    selectedRegion: undefined | string;
+    onSelectRegion: any;
+    internalCommonProps: any;
+
+    label: 'Location' | 'Region';
+    name: 'location' | 'region';
+    values: string[][];
+    href: string;
+}
+
+function prepareDefaultRegions(values: string[][]): Map<string, string> {
+    const temp = new Map<string, string>();
+    for (const [key, value] of values) {
+        temp.set(key, value);
+    }
+    return temp;
+}
+
+export default function Location(props: Props): JSX.Element {
+    const {
+        selectedRegion, onSelectRegion, internalCommonProps, name, values, href, label,
+    } = props;
+    const [regions, setRegions] = useState<Map<string, string>>(() => prepareDefaultRegions(values));
+    const [newRegionKey, setNewRegionKey] = useState<string>('');
+    const [newRegionName, setNewRegionName] = useState<string>('');
+
+    const handleAddingRegion = (): void => {
+        if (!newRegionKey || !newRegionName) {
+            notification.warning({
+                message: 'Incorrect region',
+                className: 'cvat-incorrect-add-region-notification',
+            });
+        } else if (regions.has(newRegionKey)) {
+            notification.warning({
+                message: 'This region already exists',
+                className: 'cvat-incorrect-add-region-notification',
+            });
+        } else {
+            const regionsCopy = regions;
```

Comment:
I hardly think it is a copy, that is probably link to the same object. Check if ``regions`` contains ``newRegionKey`` after it was inserted to ``regionsCopy``.

Reply:
If you need exactly copy, maybe better to use Object and create a shallow copy this way: ``const copy = { ...source }``